### ISSUE TITLE
Load data in IndexedDB from file (CSV/JSON/...)

### ISSUE THREAD

**Issue by 0live** [agershun/alasql#1075]
Hello,
I'm trying to use IndexedDB and inserting data inside from a CSV file with several thousand lines. In your documentation, you explain how to use indexed DB with a JS variable but i don't know how to use the DB after created it.

Here is my code: [JSFiddle](http://jsfiddle.net/fpe14vnd/)
When i'm doing this, i get the following message:

> alasql.js:15335 Uncaught Error: Database 'geo' does not exist

Generally we use indexed DB for heavy Dataset, is it possible to update the documentation for this case?

Thank you in advance.

**Comment by mathiasrw** [comment]
Interacting with indexedDB is async as I recall. Please try to run the first command as a promise and the second command in a .then() to see if that helps. 

Also, I recommend trying the `alasql['CREATE INDEXEDDB DATABASE...', 'ATTACH INDEXEDDB DATABASE ', 'USE geo...','...'])` notation

**Comment by 0live** [comment]
Thank you for your answer!
When i create a DB with a promise i can access it and create a table into it. Data insertion in the table seems to work too as long as i stay in the promise.
But if i reload the page, Alasql see that DB already exists, but the table is empty. 
In this [fiddle](http://jsfiddle.net/fpe14vnd/5/), i check the return of command "Create table if not exists" and it's work fine. But in line 25, when i log the tables, the "csv_extract" is empty when it should contains the data i've inserted the first time.

Do you think is it because data insertion is not working or is there another reason?

**Comment by definitivequestion** [comment]
This appears to be an issue of the database going out of scope. Is there any way to update the documentation / code for the Database constructor so we can specify its type to be INDEXEDDB? Ideally a constructor such as: Database(engine, name).

**Comment by mathiasrw** [comment]
@definitivequestion Hmmm - can you elaborate? At the moment we

    alasql('CREATE INDEXEDDB DATABASE ixbase');
    alasql('ATTACH INDEXEDDB DATABASE ixbase AS myixdb');

Are you talking about making a DB without having to write the SQL text string?

**Comment by definitivequestion** [comment]
I guess I am not understanding exactly what happens when we do something like var myDatabase = new alasql.Database('myDatabase'); I don't mind entering the create, attach, and use lines if that works. But it seems to me that creating a new database uses the memory backend and can't be changed after creation.

**Comment by mathiasrw** [comment]
You are right. You cant set the type of database via the `.Database`. Not perfect, but to be frank the `.Database` was never really meant to be consumed. 

In any case. I suggest something like 

await alasql(['CREATE INDEXEDDB DATABASE ixbase']);
await alasql(['ATTACH INDEXEDDB DATABASE ixbase']);
await alasql(['USE ixbase'])

We just found a but in the execution leading to the initial creact/attach/use not being able to be in the same wait call - so the workaround is having 3 statements...

Let me know if it works.
